org.civicrm.reqangular/test/models/contract_test.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/angularMocks',
+  'common/models/contract',
+  'common/mocks/services/hr-settings-mock',
+  'common/mocks/services/api/contract-mock',
+  'common/mocks/models/instances/contract-instance-mock'
+], function (_) {
+  'use strict';
+
+  describe('Contract', function () {
+    var $provide;
+    var $rootScope;
+    var Contract;
+    var ContractInstanceMock;
+    var contractAPI;
+    var contractAPIMock;
+    var contracts;
+
+    beforeEach(function () {
+      module('common.models', 'common.mocks', function (_$provide_) {
+        $provide = _$provide_;
+      });
+      inject([
+        'api.contract.mock', 'HR_settingsMock',
+        function (_contractAPIMock_, HRSettingsMock) {
+          contractAPIMock = _contractAPIMock_;
+          $provide.value('api.contract', contractAPIMock);
+          $provide.value('HR_settings', HRSettingsMock);
+        }
+      ]);
+    });
+
+    beforeEach(inject([
+      '$rootScope', 'Contract', 'ContractInstanceMock',
+      'api.contract',
+      function (_$rootScope_, _Contract_, _ContractInstanceMock_, _contractAPI_) {
+        $rootScope = _$rootScope_;
+
+        Contract = _Contract_;
+
+        ContractInstanceMock = _ContractInstanceMock_;
+
+        contractAPI = _contractAPI_;
+
+        contractAPI.spyOnMethods();
+
+        contracts = contractAPI.all();
```

Comment:
`contracts` are all mocked contracts. Later I am testing that `all()` returns the results with the same length that `contracts`.